src/prepare_dlc_dev_environment.py

```diff
@@ -265,114 +265,59 @@ def commit_and_push_changes(changes, remote_push=None, restore=False):
     return commit_message
 
 
-def handle_currency_option(currency_paths):
+def extract_path_components(path, pattern):
```

Comment:
for any "ERRORs" that are being printed out, make sure they return an nonzero exit code to the script user. You can accomplish this by raising an error (RuntimeError is probably the most appropriate), or exit(1) after logging the error message